e2e/content-services/sso/sso-download-directive-component.e2e.ts

```diff
@@ -0,0 +1,160 @@
+    let implicitFlow;
+    let uploadActions = new UploadActions();
+    let firstPdfFileModel = new FileModel({
+        'name': resources.Files.ADF_DOCUMENTS.PDF_B.file_name,
+        'location': resources.Files.ADF_DOCUMENTS.PDF_B.file_location
+    });
+
+    let pngFileModel = new FileModel({
+        'name': resources.Files.ADF_DOCUMENTS.PNG.file_name,
+        'location': resources.Files.ADF_DOCUMENTS.PNG.file_location
+    });
+
+    let pdfUploadedFile, pngUploadedFile, folder;
+
+    this.alfrescoJsApi = new AlfrescoApi({
+        provider: 'ECM',
+        hostEcm: TestConfig.adf.url
+    });
+    let downloadedPngFile = path.join(__dirname, 'downloads', pngFileModel.name);
+    let downloadedMultipleFiles = path.join(__dirname, 'downloads', 'archive.zip');
+    let folderName = Util.generateRandomString(5);
+
+    describe('SSO in ADF using ACS and AIS, implicit flow set', () => {
+
+        beforeAll(async (done) => {
+            await this.alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+
+            folder = await uploadActions.createFolder(this.alfrescoJsApi, folderName, '-my-');
+
+            pdfUploadedFile = await uploadActions.uploadFile(this.alfrescoJsApi, firstPdfFileModel.location, firstPdfFileModel.name, folder.entry.id);
+
+            pngUploadedFile = await uploadActions.uploadFile(this.alfrescoJsApi, pngFileModel.location, pngFileModel.name, folder.entry.id);
+
+            silentLogin = false;
+            implicitFlow = true;
+            settingsPage.setProviderEcmSso(TestConfig.adf.url, TestConfig.adf.hostSso, TestConfig.adf.hostIdentity, silentLogin, implicitFlow, 'alfresco');
+            loginSsoPage.clickOnSSOButton();
+            loginSsoPage.loginSSOIdentityService(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+
+            navigationBarPage.clickContentServicesButton();
+            contentServicesPage.checkAcsContainer();
+            contentListPage.doubleClickRow(folderName);
+            contentListPage.waitForTableBody();
+            done();
+        });
+
+        afterAll(async (done) => {
+            try {
+                await this.alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+                await uploadActions.deleteFilesOrFolder(this.alfrescoJsApi, folder.entry.id);
+            } catch (error) {
+            }
+            await this.alfrescoJsApi.logout();
+            browser.executeScript('window.sessionStorage.clear();');
+            browser.executeScript('window.localStorage.clear();');
+            done();
+        });
+
+        afterEach(async (done) => {
+            browser.refresh();
+            contentListPage.waitForTableBody();
+            done();
+        });
+
+        it('[C291936] Should be able to download a file', async (done) => {
+            contentListPage.selectRow(pngFileModel.name);
+            contentServicesPage.clickDownloadButton();
+            browser.driver.sleep(1000);
```

Comment:
is this sleep really needed?
The method Util.fileExists already waits and repeats until file is downloaded